### PR TITLE
feat : Add Fabric8 Maven Plugin deprecation log message before & after goal executions

### PR DESCRIPTION
**Pull request by rohanKanojia**
# Description

Related to https://github.com/eclipse/jkube/issues/1426

Add warning log message to print deprecation message and instruct users
to migrate to Eclipse JKube

Signed-off-by: Rohan Kumar <rohaan@redhat.com>

### CODE REVIEW

File: core/src/main/java/io/fabric8/maven/core/util/Fabric8MavenPluginDeprecationUtil.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+package io.fabric8.maven.core.util;
+
+import io.fabric8.maven.docker.util.Logger;
+
+public class Fabric8MavenPluginDeprecationUtil {
+  private Fabric8MavenPluginDeprecationUtil() { }
+
+  public static void logFabric8MavenPluginDeprecation(Logger logger, boolean logDeprecationWarning) {
+    if (logDeprecationWarning) {
+      logger.warn("Fabric8 Maven Plugin has been deprecated and would no longer be supported in future.\n\n" +
```
Comment (manusa) [inline core/src/main/java/io/fabric8/maven/core/util/Fabric8MavenPluginDeprecationUtil.java:25]:
```suggestion
      logger.warn("Fabric8 Maven Plugin has been deprecated and is no longer be supported.\n\n" +
```

File: core/src/main/java/io/fabric8/maven/core/util/Fabric8MavenPluginDeprecationUtil.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          "Please consider migrating to Eclipse JKube (https://github.com/eclipse/jkube) plugins:\n" +
+          "  - Kubernetes Maven Plugin (https://www.eclipse.org/jkube/docs/kubernetes-maven-plugin)\n" +
+          "  - OpenShift Maven Plugin (https://www.eclipse.org/jkube/docs/openshift-maven-plugin/)\n" +
+          "You can read Migration Guide for more details (https://www.eclipse.org/jkube/docs/migration-guide/)\n\n" +
```
Comment (manusa) [inline core/src/main/java/io/fabric8/maven/core/util/Fabric8MavenPluginDeprecationUtil.java:29]:
```suggestion
          "You can read the Migration Guide for more details (https://www.eclipse.org/jkube/docs/migration-guide/)\n\n" +
```

File: doc/src/main/asciidoc/inc/_introduction.adoc [hunk 3]
```diff
@@ -2,6 +2,11 @@
 [[introduction]]
 = Introduction
 
+IMPORTANT: *Fabric8 Maven Plugin has been deprecated and would no longer be supported in future.
```
Comment (manusa) [inline doc/src/main/asciidoc/inc/_introduction.adoc:5]: is no longer supported

**Comment by manusa** [comment]
@sunix could you please take a look

**Comment by sunix** [comment]
> @sunix could you please take a look
I cannot add myself as a reviewer

**Comment by manusa** [comment]
> do we want to do this now or in a next release/pr? [eclipse/jkube#287 (comment)](https://github.com/eclipse/jkube/issues/287#issuecomment-1144581566)
> 
> > Maybe what we can do is in the next release:
> > 
> > * fail when excuting one of the goals.
> > * display warnings about the problems with using fmp and provide different resolution options
> > * have a flag for user who still wants to use fmp that would reenable the goals.

That would be a period (3-6 months) after we release these changes

**Comment by manusa** [comment]
> > @sunix could you please take a look
> 
> I cannot add myself as a reviewer

:smile: yes, that's why I pinged you in a message. But you should be able to provide a review or comment an LGTM

File: core/src/test/java/io/fabric8/maven/core/util/Fabric8MavenPluginDeprecationUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+          "  - Kubernetes Maven Plugin (https://www.eclipse.org/jkube/docs/kubernetes-maven-plugin)\n" +
+          "  - OpenShift Maven Plugin (https://www.eclipse.org/jkube/docs/openshift-maven-plugin/)\n" +
+          "You can read the Migration Guide for more details (https://www.eclipse.org/jkube/docs/migration-guide/)\n\n" +
+          "Note: To disable this warning use `-Dfabric8.logDeprecationWarning=false`.");
+      times = 1;
+    }};
+  }
+
+  @Test
+  public void logFabric8MavenPluginDeprecation_whenLogDeprecationFalse_shouldLogDeprecationWarning() {
```
Comment (sunix) [inline core/src/test/java/io/fabric8/maven/core/util/Fabric8MavenPluginDeprecationUtilTest.java:45]: `..._shouldLogDeprecationWarning` or `..._shouldNotLogDeprecationWarning` ?

**Review by sunix — CHANGES_REQUESTED** [review]
Naming of the test method looks wrong.
Otherwise looks good to me.

File: doc/src/main/asciidoc/inc/_introduction.adoc [hunk 5]
```diff
@@ -2,6 +2,11 @@
 [[introduction]]
 = Introduction
 
+IMPORTANT: *Fabric8 Maven Plugin is deprecated and is no longer be supported.
```
Comment (manusa) [inline doc/src/main/asciidoc/inc/_introduction.adoc:5]:
```suggestion
IMPORTANT: *Fabric8 Maven Plugin is deprecated and is no longer supported.
```

**Review by manusa — APPROVED** [review]

**Pull request merged**